puppetboard/app.py

```diff
@@ -1109,3 +1110,20 @@ def radiator(env):
         stats=stats,
         total=num_nodes
     )
+
+@app.route('/daily_reports_chart.json', defaults={'env': app.config['DEFAULT_ENVIRONMENT']})
+@app.route('/<env>/daily_reports_chart.json')
+def daily_reports_chart(env):
+    """Return JSON data to generate a bar chart of daily runs.
+
+    If certname is passed as GET argument, the data will target that
+    node only.
+    """
+    certname = request.args.get('certname')
+    result = get_daily_reports_chart(
```

Comment:
result = get_or_abort(get_daily_reports_chart, ...)
This will allow for nice clean exit if you get a failure calling query()